tests/framework/integration/cluster.go

```diff
@@ -1410,32 +1405,33 @@ func (c *Cluster) Endpoints() []string {
 	return endpoints
 }
 
-func (c *Cluster) ClusterClient() (client *clientv3.Client, err error) {
-	if c.clusterClient == nil {
-		var endpoints []string
-		for _, m := range c.Members {
-			endpoints = append(endpoints, m.GrpcURL)
-		}
-		cfg := clientv3.Config{
-			Endpoints:          endpoints,
-			DialTimeout:        5 * time.Second,
-			DialOptions:        []grpc.DialOption{grpc.WithBlock()},
-			MaxCallSendMsgSize: c.Cfg.ClientMaxCallSendMsgSize,
-			MaxCallRecvMsgSize: c.Cfg.ClientMaxCallRecvMsgSize,
-		}
-		if c.Cfg.ClientTLS != nil {
-			tls, err := c.Cfg.ClientTLS.ClientConfig()
-			if err != nil {
-				return nil, err
-			}
-			cfg.TLS = tls
-		}
-		c.clusterClient, err = newClientV3(cfg)
+func (c *Cluster) ClusterClient(opts ...func(*clientv3.Config)) (client *clientv3.Client, err error) {
```

Comment:
I don't think it's good to have inconsistency between `ClusterClient` and `Client` methods. Could update the integration. Cluster code so it also doesn't require user to call `Close`.

Reply:
In order to achieve that, I need to register the test clean up function whenever `ClusterClient` is called like this

```
	t.Cleanup(func() {
		cc.Close()
	})
```

However, `testing.TB` is not stored in the https://github.com/etcd-io/etcd/blob/cd9764a99fab81788d6234d3524a08c9b893c3e4/tests/framework/integration/cluster.go#L177-L184

Do you think one of the implementation can be adding this to the Cluster struct? Thanks!

Reply:
I think it's ok to add `testing.TB` to integration.Cluster.